cetmix_tower_server/tests/test_plan.py

```diff
@@ -322,16 +322,50 @@ def test_plan_user_access_rule(self):
         )
         self.assertTrue(self.plan_3)
 
-        # Check what manager can't unlink exisiting plan
+        # Check what manager can't unlink existing plan
         with self.assertRaises(AccessError):
             self.plan_3.with_user(self.user_bob).unlink()
 
         # Add user_bob to group_root
         self.add_to_group(self.user_bob, "cetmix_tower_server.group_root")
 
-        # Check what root can unlink exisiting plan
+        # Check what root can unlink existing plan
         result = self.plan_3.with_user(self.user_bob).unlink()
         self.assertTrue(result)
+        # Remove user_bob from root group to test access to flight plans related
+        # to servers user isn't subscribed
+        self.remove_from_group(
+            self.user_bob,
+            [
+                "cetmix_tower_server.group_root",
+            ],
+        )
+        # Assign plan_1 to self.server_test_1
+        self.plan_2.write({"server_ids": [(6, 0, [self.server_test_1.id])]})
+        # Ensure Bob can't access plan as manager if he is not a follower
+        #  of self.server_test_1
+        with self.assertRaises(AccessError):
+            self.plan_2.with_user(self.user_bob).read([])
+        # Subscribe Bob to self.server_test_1
+        self.server_test_1.message_subscribe([self.user_bob.partner_id.id])
+        # Ensure Bob can now access plan as a follower of self.server_test_1
+        plan_2_read_result = self.plan_2.with_user(self.user_bob).read([])
+        self.assertEqual(
+            plan_2_read_result[0]["name"],
+            self.plan_2.name,
+            msg="Plan name should be same",
+        )
+        # Remove Bob from manager group
+        self.remove_from_group(self.user_bob, ["cetmix_tower_server.group_manager"])
+        # Change plan access level
+        self.plan_2.write({"access_level": "1"})
```

Comment:
Please use [wirte_and_invalidate](https://github.com/cetmix/cetmix-tower/blob/e70e241b99cd7bfdc27e389be935b2e2eef34db9/cetmix_tower_server/tests/common.py#L330) method to ensure caches are invalidated.

Reply:
Thanks for the suggestion. I've updated the code.